stacker/tests/hooks/test_lambda.py

```diff
@@ -290,11 +288,29 @@ def test_idempotence(self):
                 code = results.get('MyFunction')
                 self.assertIsInstance(code, Code)
 
-                info = self.s3.head_object(Bucket=code.S3Bucket,
-                                           Key=code.S3Key)
-                if not version:
-                    version = info['VersionId']
-                else:
-                    compare(version, info['VersionId'],
-                            prefix='S3 object must not be modified in '
-                                   'repeated runs')
+                if not previous:
+                    previous = code.S3Key
+                    continue
+
+                compare(previous, code.S3Key,
+                        prefix="zipfile name should not be modified in "
+                               "repeated runs.")
+
+    def test_calculate_hash(self):
+        with self.temp_directory_with_files() as d1:
+            root = d1.path
+            hash1 = _calculate_hash(ALL_FILES, root)
+
+        with self.temp_directory_with_files() as d2:
+            root = d2.path
+            hash2 = _calculate_hash(ALL_FILES, root)
+
+        with self.temp_directory_with_files() as d3:
+            root = d3.path
+            with open(os.path.join(root, ALL_FILES[0]), "w") as fd:
+                fd.write("modified file data")
+            hash3 = _calculate_hash(ALL_FILES, root)
+
```

Comment:
It would be nice to add a test verifying that the same set of files with the same content, but different file names generate a different hash.